compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Repl.hs

```diff
@@ -191,54 +210,73 @@ runRepl opts mainDar replClient ideState = do
             initialiser = pure ()
     State.evalStateT replM initReplState
   where
-    handleLine
-        :: [LF.ModuleName]
-        -> [(LPat GhcPs, Type)]
-        -> DynFlags
+    handleStmt
+        :: DynFlags
         -> String
-        -> Int
-        -> IO (Either Error (LPat GhcPs, Type))
-    handleLine moduleNames binds dflags l i = runExceptT $ do
-        stmt <- case parseStatement l dflags of
-            POk _ lStmt -> pure (unLoc lStmt)
-            PFailed _ _ errMsg -> throwError (ParseError errMsg)
-        (bind, expr) <- maybe (throwError (UnsupportedStatement l)) pure (splitStmt stmt)
-        liftIO $ writeFileUTF8 (fromNormalizedFilePath $ lineFilePath i)
-            (renderModule dflags moduleNames i binds bind expr)
+        -> Stmt GhcPs (LHsExpr GhcPs)
+        -> ExceptT Error ReplM ()
+    handleStmt dflags line stmt = do
+        ReplState {imports, bindings, lineNumber} <- State.get
+        (bind, expr) <- maybe (throwError (UnsupportedStatement line)) pure (splitStmt stmt)
+        liftIO $ writeFileUTF8 (fromNormalizedFilePath $ lineFilePath lineNumber)
+            (renderModule dflags imports lineNumber bindings bind expr)
         -- Useful for debugging, probably best to put it behind a --debug flag
-        -- rendered <- liftIO  $readFileUTF8 (fromNormalizedFilePath $ lineFilePath i)
+        -- rendered <- liftIO $ readFileUTF8 (fromNormalizedFilePath $ lineFilePath lineNumber)
```

Comment:
We should probably just add that debugging flag :slightly_smiling_face:  (separately)